### PR TITLE
Sprint02/restructure/backend/restructuring groups

### PR DESCRIPTION
**Pull request by pabloerhard**

### CODE REVIEW

File: backend/src/modules/group/groupController.ts
Comment (NicoHwpo) [inline backend/src/modules/group/groupController.ts:1]: missing swagger yaml for all endpoints...

File: backend/src/modules/group/groupRepository.ts [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+                id: group.id,
+                name: group.name,
+                area: group.area ?? '',
+                createdAt: group.createdAt,
+                updatedAt: group.updatedAt ?? new Date(),
+            };
+        } catch (error) {
+            if (error instanceof Prisma.PrismaClientKnownRequestError) {
+                if (error.code === 'P2002') {
+                    throw new CustomError(409, 'Role already exists');
```
Comment (NicoHwpo) [inline backend/src/modules/group/groupRepository.ts:27]: "Role alreay exists" ??? when creating a group? hmmmmm. Should be "Group already exists"

File: backend/src/modules/group/groupController.ts
Comment (pabloerhard) [reply backend/src/modules/group/groupController.ts:1]: Hi Nico, I hope you are doing great! Unfortunately swagger endpoints weren't included in my tasks for the past sprint. Are you inferring im supposed to work overtime? If yes we can probably renegotiate my current contract. Thank you!

File: backend/src/modules/group/groupRepository.ts
Comment (NicoHwpo) [inline backend/src/modules/group/groupRepository.ts:1]: missing return value types... Not checking for possible errors

File: backend/src/modules/group/groupRepository.ts (hunk 1 above)
Comment (pabloerhard) [reply backend/src/modules/group/groupRepository.ts:27]: It should be Group already exists, this is a copy and paste error.

File: frontend/src/app/admin/components/side-nav.tsx
Comment (NicoHwpo) [inline frontend/src/app/admin/components/side-nav.tsx:1]: why changes on the frontend?

Comment (pabloerhard) [reply frontend/src/app/admin/components/side-nav.tsx:1]: Not my changes, probably originated from merge conflict fix.

**Review by NicoHwpo — CHANGES_REQUESTED** [review]
check comments on code...
- Add returning types for functions on the repository...
- add error handling... dont return a prisma function...
    - call the function in a variable, and then check if the variable is null (and if it is, return custom error)
- add yaml swagger code to all groups endpoints... (search how to or ask chatgpt)

File: backend/src/modules/group/groupController.ts
Comment (NicoHwpo) [reply backend/src/modules/group/groupController.ts:1]: Swagger documentation is always implied and required for any endpoint. That is some basic knowledge for backend development and for your position at this company.
And sure, we can have a quick meeting to review your contract. I suggest bringing your CV to the meeting so we can check again if your current skills align with your current role... or if you are lying on your resume.


Best,
Nico

Comment (pabloerhard) [reply backend/src/modules/group/groupController.ts:1]: I appreciate your emphasis on Swagger documentation. It's important to recognize that Swagger's implementation was introduced late in the project and may not be achievable for this sprint. Furthermore, it's worth noting that while Swagger is valuable, it's not universally integrated into all backend projects.

Furthermore, it's worth noting that Swagger wasn't included in our initial requirements. This implies that it was introduced after the delivery of our requirements. While requirements can evolve, the introduction of Swagger as a project component occurred after our initial specifications were delivered. This suggests that it should not be implied or considered a mandatory requirement for every endpoint in a backend project.

Regarding the meeting I reject attending.

Worst, 
Pablo

**Review by NicoHwpo — APPROVED** [review]
looks alr ig.

**Pull request merged**